WalletWasabi.Gui/Tabs/EncryptionManager/DecryptMessageViewModel.cs

```diff
@@ -0,0 +1,221 @@
+{
+	internal class DecryptMessageViewModel : CategoryViewModel, IDisposable
+	{
+		private string _encryptedMessage;
+		private string _password;
+		private string _decryptedMessage;
+		private string _myPublicKey;
+		private string _warningMessage;
+		private bool _isPublicKeyPresent;
+		private bool _isSearchResultExpanded;
+		private ObservableCollection<AddressPubKeyViewModel> _addresses;
+		private AddressPubKeyViewModel _selectedItem;
+		private IEnumerable<AddressPubKeyViewModel> _allKeysViewModels;
+		private string _addressSearch;
+		private CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+		public string EncryptedMessage
+		{
+			get => _encryptedMessage;
+			set => this.RaiseAndSetIfChanged(ref _encryptedMessage, value);
+		}
+
+		public string Password
+		{
+			get => _password;
+			set => this.RaiseAndSetIfChanged(ref _password, value);
+		}
+
+		public string MyPublicKey
+		{
+			get => _myPublicKey;
+			set => this.RaiseAndSetIfChanged(ref _myPublicKey, value);
+		}
+
+		public string DecryptedMessage
+		{
+			get => _decryptedMessage;
+			set => this.RaiseAndSetIfChanged(ref _decryptedMessage, value);
+		}
+
+		public string AddressSearch
+		{
+			get => _addressSearch;
+			set => this.RaiseAndSetIfChanged(ref _addressSearch, value);
+		}
+
+		public string WarningMessage
+		{
+			get => _warningMessage;
+			set => this.RaiseAndSetIfChanged(ref _warningMessage, value);
+		}
+
+		public bool IsPublicKeyPresent
+		{
+			get => _isPublicKeyPresent;
+			set => this.RaiseAndSetIfChanged(ref _isPublicKeyPresent, value);
+		}
+
+		public bool IsSearchResultExpanded
+		{
+			get => _isSearchResultExpanded;
+			set => this.RaiseAndSetIfChanged(ref _isSearchResultExpanded, value);
+		}
+
+		public AddressPubKeyViewModel SelectedItem
+		{
+			get => _selectedItem;
+			set => this.RaiseAndSetIfChanged(ref _selectedItem, value);
+		}
+
+		public ObservableCollection<AddressPubKeyViewModel> Addresses
+		{
+			get => _addresses;
+			set => this.RaiseAndSetIfChanged(ref _addresses, value);
+		}
+
+		public ReactiveCommand DecryptCommand { get; }
+		public ReactiveCommand MyPublicKeyCommand { get; }
+
+		public EncryptionManagerViewModel Owner { get; }
+
+		public DecryptMessageViewModel(EncryptionManagerViewModel owner) : base("Decrypt Message")
+		{
+			Owner = owner;
+
+			var canDecrypt = this.WhenAnyValue(x => x.EncryptedMessage,
+				(message) =>
+					!string.IsNullOrEmpty(message));
+
+			DecryptCommand = ReactiveCommand.Create(
+				() =>
+				{
+					WarningMessage = "";
+					DecryptedMessage = DecryptMessage(EncryptedMessage, MyPublicKey, Password);
+				},
+				canDecrypt
+			);
+
+			DecryptCommand.ThrownExceptions.Subscribe(ex =>
+			{
+				ShowWarnMessage(ex.Message);
+			});
+
+			MyPublicKeyCommand = ReactiveCommand.Create(
+				() =>
+				{
+					WarningMessage = "";
+					DecryptedMessage = DecryptMessage(EncryptedMessage, MyPublicKey, Password);
```

Comment:
I think we should catch the exception and display a human understandable message here, otherwise it will fail with erros that only cryptographers will understand.

Reply:
Decryption failed. Please check the correctness of Encrypted message, Public Key and Password! Details: ex.Message
I agree but I will write the exact error message after the details.